File: cli/clients/oikos.ts
// SPDX-License-Identifier: PMPL-1.0

import { postJson } from "../http.ts";
import type { HttpOptions } from "../types.ts";

/** Score a package for sustainability with oikos service */
export async function score(
  packageName: string,
  baseUrl: string,
  token: string | undefined,
  opts: HttpOptions,
): Promise<void> {
  const body = JSON.stringify({ package: packageName });
  await postJson(baseUrl, "/score", token, body, opts);
}

File: crates/opm-cli/src/clients/oikos.rs
// SPDX-License-Identifier: PMPL-1.0

use anyhow::Result;

use crate::clients::http;

pub fn score(package: &str, base_url: &str, token: Option<&str>, opts: &http::HttpOptions) -> Result<()> {
    let body = format!("{\"package\":\"{package}\"}");
    http::post_json(base_url, "/score", token, &body, opts)
}
